File: python/linearsearch.py
# Linear search with recursion

def search(arr, target, ind = 0):
    if ind == len(arr): return -1
    elif arr[ind] == target: return ind
    else: return search(arr, target, ind + 1)

def searchLast(arr, target, ind):
    if ind == -1: return -1
    elif arr[ind] == target: return ind
    else: return searchLast(arr, target, ind - 1)

print(search([1, 2, 3, 4, 5], 0))
print(searchLast([1, 2, 3, 1, 3, 5, 8], 1, 6))

File: c/linearsearch.c
// Linear search using recursion

#include <stdio.h>

// Search for the first appearance of a target in array using linear search through recursion
int search(int *arr, int arrSize, int target, int ind);
// Search for the last appearance of a target in array using linear search through recursion
int searchLast(int *arr, int arrSize, int target, int ind);

int main() {
    int arrSize = 10,
        arr[] = {1, 2, 3, 4, 5, 6, 7, 4, 9, 10},
        target = 4;

    printf("%d\n", search(arr, arrSize, target, 1));
    printf("%d\n", searchLast(arr, arrSize, target, arrSize - 1));

    return 0;
}

int search(int *arr, int arrSize, int target, int ind) {
    if (ind == arrSize) return -1;
    else if (arr[ind] == target) return ind;
    else return search(arr, arrSize, target, ind + 1);
}

int searchLast(int *arr, int arrSize, int target, int ind) {
    if (ind == -1) return -1;
    else if (arr[ind] == target) return ind;
    else return searchLast(arr, arrSize, target, ind - 1);
}